### ISSUE TITLE
`var-annotated` behaves differently with daemon

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#10847]
**Bug Report**

```py
class Foo:
    bar = None
```

when running with `dmypy`:

```
$ dmypy run -- foo.py
Restarting: configuration changed
Daemon stopped
Daemon started
foo.py: note: In class "Foo":
foo.py:2: error: Need type annotation for "bar"  [var-annotated]
        bar = None
        ^
Found 1 error in 1 file (checked 1 source file)
```

when running with `mypy`:
```
$ mypy foo.py
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files):
  ```ini
  [mypy]
  exclude = venv
  mypy_path = stubs
  warn_redundant_casts = True
  show_error_context = True
  pretty = True
  follow_imports = skip
  allow_redefinition = True
  show_error_codes = True
  ```
- Python version used: 3.9.6
- Operating system and version: windows 10 1909

**Comment by KotlinIsland** [comment]
Isn't this just because the daemon implicitly enables `local_partial_types`?
https://mypy.readthedocs.io/en/stable/config_file.html#confval-local_partial_types

**Comment by DetachHead** [comment]
seems silly that the daemon secretly changes random flags, i would expect it to behave the same way

**Comment by hauntsaninja** [comment]
Yeah, this is because of local partial types. I think Jukka is planning on making that the default, so closing as duplicae of https://github.com/python/mypy/pull/10169

**Comment by KotlinIsland** [comment]
It's the default in basedmypy

**Comment by vergenzt** [comment]
Found this issue from googling, and sharing my further digging:
- it's turned on at https://github.com/python/mypy/blob/55d4c1725bae29ad5ac2ce857b4b4b3363e5518c/mypy/dmypy_server.py#L197-L199
- reasoning for the flag being turned on in daemon mode is because otherwise the daemon is buggy: see https://github.com/python/mypy/issues/4492
- in future, this error is going to be default for CLI invocations too: https://github.com/python/mypy/pull/10169 (but it's not yet)

**Comment by KotlinIsland** [comment]
> in future, this error is going to be default for CLI invocations too

it's the default in basedmypy
